lib/cluster/vpc.rb

```diff
@@ -0,0 +1,68 @@
+require 'pry'
+module Cluster
+  class VPC < Base
+    def self.all
+      vpcs = []
+      ec2.describe_vpcs.each do |page|
+        page.vpcs.each do |vpc|
+          vpcs << vpc
+        end
+      end
+      vpcs
+    end
+
+    def self.create_or_initialize
+      vpc = find_vpc
+      return vpc if vpc
+
+      if requested_vpc_has_conflicts_with_existing_one?
+        raise VpcConflictsWithAnother
+      else
+        vpc = ec2.create_vpc(
+          cidr_block: vpc_config[:cidr_block]
+        ).first.vpc
+        vpc_instance = Aws::EC2::Vpc.new(vpc.vpc_id, client: ec2)
+        vpc_instance.create_tags(
+          tags: [{ key: 'Name', value: vpc_config[:name] }]
+        )
+      end
+
+      vpc
+    end
+
+    private
+
+    def self.ec2
+      @@ec2 ||= Aws::EC2::Client.new(
+        region: config.json[:region],
+        credentials: config.credentials
+      )
+    end
+
+    def self.requested_vpc_has_conflicts_with_existing_one?
+      vpc = all.find do |vpc|
+        (vpc.cidr_block == vpc_config[:cidr_block]) ||
```

Comment:
Fixed, though the way this private method found an existing VPC first (by matching on name AND cidr_block) meant we probably wouldn't have ever hit this.